dbms/src/Flash/EstablishCall.cpp

```diff
@@ -202,6 +202,11 @@ void EstablishCallData::trySendOneMsg()
     switch (async_tunnel_sender->pop(res, this))
     {
     case GRPCSendQueueRes::OK:
+        /// Note: has to switch the memory tracker before `write`
+        /// because after `write`, `async_tunnel_sender` can be destroyed at any time
+        /// so there is a risk that `res` is destructed after `aysnc_tunnel_sender`
+        /// is destructed which may cause the memory tracker in `res` become invalid
+        res->switchMemTracker(nullptr);
```

Comment:
It may cause MemTracker incorrect... How about switching to RootMemTracker?

Reply:
Why? It just release the memory before this change, the memory is also released after `write()`

Reply:
I'm worried about an extreme case that a lot of threads are doing `EstablishCallData::trySendOneMsg()` and early freeing the MemTracker cause real memory usage too large than tracked

Reply:
But in before this pr, it also has the early free issue since async tunnel's write actually does not wait until write finish.